full-stack-tests/backend/src/integration/CloudSqlite.test.ts

```diff
@@ -188,6 +191,90 @@ describe("CloudSqlite", () => {
+    await azSqlite.subtractFromCurrentWriteLockExpiryTime(testContainer1, "bcv_kv.bcv", 5);
+    // user2 grabs the write lock
+    testContainer2.connect(testCache2);
+    testContainer2.acquireWriteLock(user2);
+    // user1 tries to upload changes, it should fail because user1's write lock has expired and user2 is using it
+    // eslint-disable-next-line @typescript-eslint/promise-function-async
+    expect(() => testContainer1.uploadChanges()).to.throw(`Container [${testContainer1.containerId}] is currently locked by another user`);
+    // user 2 make some changes and release the write lock
+    await testContainer2.copyDatabase("testBim", "testBimCopy1");
+    const db2 = await BriefcaseDb.open({ fileName: "testBimCopy1", container: testContainer2 });
+    db2.saveFileProperty({ name: "upload", namespace: "uploadTest", id: 1, subId: 1 }, "this is a test");
+    db2.close();
+    testContainer2.releaseWriteLock();
+    // eslint-disable-next-line @typescript-eslint/promise-function-async
+    expect(() => testContainer1.uploadChanges()).to.throw(`Container [${testContainer1.containerId}] is currently locked by another user`);
+    testContainer1.disconnect();
+    testContainer2.disconnect();
+  });
+
+  it("Should refresh write lock for the current user when uploading changes", async () => {
+    const testContainer1 = testContainers[3];
+    const testCache1 = azSqlite.makeCaches(["testCache1"])[0];
+    testContainer1.connect(testCache1);
+    let currentTime = new Date();
+    testContainer1.acquireWriteLock(user1);
+    await testContainer1.copyDatabase("testBim", "testBimCopy1");
+    const db1 = await BriefcaseDb.open({ fileName: "testBimCopy1", container: testContainer1 });
+    db1.saveFileProperty({ name: "refresh", namespace: "refreshTest", id: 1, subId: 1 }, "this is a test");
+    db1.close();
+    // case 1: current write lock has expired
+    // mock 5 min operation time by user1
+    // write lock is expired after 5 min
+    await azSqlite.subtractFromCurrentWriteLockExpiryTime(testContainer1, "bcv_kv.bcv", 5);
+    // there are no other users, uploadChanges should refresh the write lock for the current user
+    await testContainer1.uploadChanges();
+    // should be 5 min but let's use 4.5 min to avoid time conflict
+    expect (await azSqlite.isWriteLockValidForAtLeast(testContainer1, "bcv_kv.bcv", currentTime, 4.5*60*1000)).to.be.true;
+    // case 2: current write lock has not expired
+    currentTime = new Date();
+    // the write lock is refreshed to 5 minutes later
+    // mock 4 min operation time by user1
+    await azSqlite.subtractFromCurrentWriteLockExpiryTime(testContainer1, "bcv_kv.bcv", 4);
+    // now user1 only has 1 min to perform further actions
+    // uploadChanges does nothing this time but to refresh write lock
+    await testContainer1.uploadChanges();
+    // should be 5 min but let's use 4.5 min to avoid time conflict
```

Comment:
Since we had 1 min left we're actually adding 5 mins to get up to 6 mins right?

Reply:
No, as long as user `uploadChanges`, `AcquireWriteLock()` refreshes the write lock expiry time to (SERVER_TIME+ 5 mins). In other words, the new expiry time is based on server time, not the previous expiry time.

So after the refresh, users should be able to own write lock for another 5 mins instead of 6 mins